### PR TITLE
fix(remotebuild): use full project name in log files

### PR DESCRIPTION
**Pull request by mr-cal**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

Use the full project name in log files.

Fixes https://github.com/canonical/snapcraft/issues/4781

### CODE REVIEW

File: craft_application/services/remotebuild.py [hunk 1]
```diff
@@ -185,7 +184,7 @@ def fetch_logs(self, output_dir: pathlib.Path) -> Mapping[str, pathlib.Path | No
             if not url:
                 logs[build.arch_tag] = None
                 continue
-            filename = f"{project_name}_{build.arch_tag}-{fetch_time}.txt"
+            filename = f"{self._app.name}_{build.arch_tag}_{fetch_time}.txt"
```
Comment (lengau) [inline craft_application/services/remotebuild.py:187]:
```suggestion
            filename = f"{self._name}_{build.arch_tag}_{fetch_time}.txt"
```

Did you mean this?

Comment (mr-cal) [reply craft_application/services/remotebuild.py]: :facepalm: yes, thanks

**Review by tigarmo — APPROVED** [review]

File: craft_application/services/remotebuild.py [hunk 2]
```diff
@@ -185,7 +184,7 @@ def fetch_logs(self, output_dir: pathlib.Path) -> Mapping[str, pathlib.Path | No
             if not url:
                 logs[build.arch_tag] = None
                 continue
-            filename = f"{project_name}_{build.arch_tag}-{fetch_time}.txt"
+            filename = f"{self._name}_{build.arch_tag}_{fetch_time}.txt"
```
Comment (dariuszd21) [inline craft_application/services/remotebuild.py:187]: Changing delimiter before `fecht_time` seems like incompatible change for me.
Is it required?

Comment (lengau) [reply craft_application/services/remotebuild.py:187]: Since remote-build is still marked as experimental I think we're fine to do this.

**Review by lengau — APPROVED** [review]

**Review by dariuszd21 — APPROVED** [review]

**Pull request merged**